src/plugins/mer/meremulatordeviceoptions.h

```diff
@@ -0,0 +1,76 @@
+#ifndef MEREMULATORDEVICEOPTIONS_H
+#define MEREMULATORDEVICEOPTIONS_H
+
+#include <QDataStream>
+
+#include <QMetaEnum>
+
+namespace Mer {
+
+namespace Internal {
+
+class MerEmulatorDeviceOptions {
+    Q_GADGET
+
+public:
+
+    /*
+     * NOTE: You are able to add new recognisable data members to OptionType
+     * Enum variables should exactly match values in MerDeviceModel.DconfDb key from mersdk-device-models.xml
+     */
+    enum OptionType {
```

Comment:
It shouldn't be limited to a hard-coded set of options. It should be possible to set any value as 1) new device models may need to set values not considered before and 2) available values may vary in different Sailfish OS versions. The one who needs to use this feature need to figure somehow all the values to be set - the easiest way to do it is to copy the dconf db snippet from the respective device adaptation sources. So a text edit would be sufficient and more suitable for this purpose.

Reply:
Yes, we are talking about devices. The purpose of setting these options is to let the UI on emulator look exactly like on some particular device. For that reason you need to copy the values from the particular device adaptation package.

Reply:
@sergey-levin @martyone 

I propose to proceed as follows.
1. Add a text area in which the options ​​for dconf are set in rows.
   UX should be the same as in Options -> Testing -> General -> Global Filters.
2. Save individual controls for the most common settings, as is done now.
3. Make the autosynchronization of values ​​in the controls and the text area.

Additionally, I propose to to hide text area under the "Details" button, since (I hope) the manual fine tuning of dconf (provided by device profile) will not be required in most cases.
At the same time, it should be an option for importing and exporting a device profile to a file.

Reply:
@chuvilin It would look nice, indeed, but above that I do not see any value it would add. I have been updating the models couple of times already and all I needed was to paste a code snippet somewhere, nothing more. Like in Options > Text Editor > Snippets

Reply:
@martyone 
Perhaps it's better to have a general text area instead of line-based editor to be able to paste a code snippet as you say. Could you please describe the source of such snippets and the format?

But to have a Text Editor only is a good case when you know all the names and actual values of dconf arguments or have such a snippet. I prefer to leave controls that represent the main values since they are already done.